shortcode.php

```diff
@@ -330,3 +330,110 @@ function geoip_detect_shortcode_user_info($attr) {
     return geoip_detect2_shortcode_user_info_wpcf7('', 'geoip_detect2_user_info', true);
 }
 add_shortcode('geoip_detect2_user_info', 'geoip_detect_shortcode_user_info');
+
+/**
+ *
+ * Content Hiding
+ * 
+ * Uses an enclosing shortcode (i.e. [shortcode]Content[/shortcode]). Shortcode attributes can be as follows:
+ * "timezone", "continent", "country", "region"/"state", "city", and "and-not".
+ * 
+ * Each attribute may only appear once in a shortcode, and each attribute can only contain a single value, or
+ * the shortcode will break, showing the contents no matter what.
+ *
+ * Examples:
+ *
+ * `[geoip_detect2_hide_unless country="US" state="TX"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * 	    - OR -
+ * `[geoip_detect2_hide_unless country="US" region="TX"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * 	    - OR -
+ * `[geoip_detect2_hide_unless country="US" region="Texas"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's country and region/state.
+ * 
+ * `[geoip_detect2_hide_unless country="US" state="TX" and-not="Houston"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's country and region/state,
+ * and the specified exclusion is not found in any possible GeoIP Detect field (e.g. visitor's country,
+ * city, timezone, etc.).
+ * 
+ * `[geoip_detect2_hide_unless contenant="North America"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's contenant.
+ * 
+ */
+function geoip_detect2_shortcode_hide_unless($atts, $content = null) {
+    $atts_array = shortcode_atts(array(
+        'timezone' => null,
+        'continent' => null,
+        'country' => null,
+        'region' => null,
+        'state' => null,
+        'city' => null,
+        'and-not' => null),
+        $atts);
+
+    $info = geoip_detect2_get_info_from_current_ip();
```

Comment:
Please add `skip_cache` and `lang` as in `geoip_detect2_shortcode()` ...

Reply:
I will need to look into this later. (This is the first time I have ever attempted to write anything in PHP.)